src/server/req_preemptjob.c

```diff
@@ -72,6 +71,199 @@ extern struct server server;
 
 extern time_t time_now;
 
+static int preempt_total = 0;
+static int preempt_index = 0;
+static preempt_job_info *preempt_jobs_list = NULL;
+
+/**
+ * @brief mark a job preemption as failed
+ * @param[in] preempt_preq - the preemption preq from the scheduler
+ * @param[in] job_id - the job to mark as failed
+ * @return void
+ */
+static void job_preempt_fail(struct batch_request *preempt_preq, char *job_id) {
+	preempt_preq->rq_reply.brp_code = 1;
+	strcpy(preempt_jobs_list[preempt_index].order, "000");
+	sprintf(preempt_jobs_list[preempt_index].job_id, "%s", job_id);
+	preempt_index++;
+	log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, job_id, "Job failed to be preempted");
+}
+
+/**
+ * @brief create a local batch_request for a suspend request
+ * @param[in] job_id - the job to create the request for
+ */ 
+static struct batch_request *create_suspend_request(char *job_id) 
+{
+	struct batch_request *newreq;
+
+	newreq = alloc_br(PBS_BATCH_SignalJob);
+	if (newreq == NULL) 
+		return NULL;
+	snprintf(newreq->rq_ind.rq_signal.rq_jid, sizeof(newreq->rq_ind.rq_signal.rq_jid), "%s", job_id);
+	snprintf(newreq->rq_ind.rq_signal.rq_signame, sizeof(newreq->rq_ind.rq_signal.rq_signame), "%s", "suspend");
+	return newreq;
+}
+
+/**
+ * @brief create a local batch_request for a checkpoint request
+ * @param[in] job_id - the job to create the request for
+ */
+
+static struct batch_request *create_ckpt_request(char *job_id) 
+{
+	int hold_name_size;
+	int hold_val_size = 2; /* 2 for 's' and '\0' */
+	struct batch_request *newreq;
+	svrattrl *hold_svrattrl;
+
+	hold_name_size = strlen(job_attr_def[(int)JOB_ATR_hold].at_name) + 1;
+	newreq = alloc_br(PBS_BATCH_HoldJob);
+	hold_svrattrl = attrlist_alloc(hold_name_size, 0, hold_val_size);
+	if (newreq == NULL || hold_svrattrl == NULL) {
+		if (newreq != NULL)
+			free_br(newreq);
+		if (hold_svrattrl != NULL)
```

Comment:
Since there are two, it could be an else if.  I'd rather leave it alone because if in the future we add a third variable we need to allocate for, we'll need to make it a series of ifs then.  If you really want it an else if, I'll do it.